frame/support/procedural/src/pallet/mod.rs

```diff
@@ -0,0 +1,49 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2020 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Implementation for pallet attribute macro.
+//!
+//! General workflow:
+//! 1 - parse all pallet attributes:
+//!   This step remove all attributes `#[pallet::*]` from the ItemMod and build the `Def` struct
+//!   which holds the ItemMod without `#[pallet::*]` and information given by those attributes
+//! 2 - expand from the parsed information
+//!   This step will modify the ItemMod by adding some derive attributes or phantom data variants
+//!   to user defined types. And also crate new types and implement block.
+
+mod parse;
+mod expand;
+
+pub use parse::Def;
+use syn::spanned::Spanned;
+
+pub fn pallet(
+	attr: proc_macro::TokenStream,
+	item: proc_macro::TokenStream
+) -> proc_macro::TokenStream {
+	if !attr.is_empty() {
+		let msg = "Invalid pallet macro call: expect `#[frame_support::pallet]`";
```

Comment:
The expect message doesnt immediately make sense to me. if someone types `#[pallet]` you will tell them to type `#[frame_support::pallet]` instead?

Reply:
indeed I improved error message, the span make it also more meaningfull, see ui test https://github.com/paritytech/substrate/pull/6877/commits/cc165a36aa85006f6089cc44ac70135948e69b2b